### PR TITLE
Fix issue reseting price

### LINKED ISSUE #844: Issue updating price (reseting old price)
**Description**

It doesn't break anything because the the address is not being used if there's no amount on the price, however it's technically incorrect.

**How to Reproduce in production**
Go to production, and enable redux debugging. Then observe that when we reset the price to invalidate the previous one, we set the wrong address. The estimation will be in DAI in this example, not in WETH.

image

We can see how this is not an important issue, because when we set back the new price, we also update the token to the right value:
image

**Expected behavior**
It should have the right address depending if the order is a sell or a buy order

### PR DESCRIPTION
**Pull request by anxolin**
Closes #844

Great catch from Leandro that escaped our eye!
https://github.com/gnosis/cowswap/pull/809#discussion_r661624922

### CODE REVIEW

File: src/custom/state/price/reducer.ts [hunk 1]
```diff
@@ -56,7 +57,8 @@ function initializeState(
 function getResetPrice(sellToken: string, buyToken: string, kind: string) {
   return {
     amount: null,
-    token: kind ? sellToken : buyToken
+    // When we buy, the price estimation is given in sell tokens (if we sell, we give it in sell tokens)
+    token: kind === OrderKind.BUY ? sellToken : buyToken
```
Comment (W3stside) [inline src/custom/state/price/reducer.ts:61]: the comment is confusing.

but this is: for buy orders, we want to reset the price for the sellToken, and for sell orders we want to reset the price for the buyToken

right?

**Review by W3stside — APPROVED** [review]
commenting on the comment in my comment commenting on the comment about the kind which is commented on line 60

File: src/custom/state/price/reducer.ts [hunk 2]
```diff
@@ -56,7 +57,8 @@ function initializeState(
 function getResetPrice(sellToken: string, buyToken: string, kind: string) {
```
Comment (alfetopito) [inline src/custom/state/price/reducer.ts:57]:
```suggestion
function getResetPrice(sellToken: string, buyToken: string, kind: OrderKind) {
```

File: src/custom/state/price/reducer.ts (hunk 1 above)
Comment (alfetopito) [reply src/custom/state/price/reducer.ts:61]: I second this.

Comment (anxolin) [reply src/custom/state/price/reducer.ts:64]: done 👌

File: src/custom/state/price/reducer.ts (hunk 2 above)
Comment (anxolin) [reply src/custom/state/price/reducer.ts:57]: Okok, let me do this and some rearangements in another PR, cause i think this comes from a bad modeling we did.

Comment (anxolin) [reply src/custom/state/price/reducer.ts:57]: #902

**Pull request merged**